src/main/java/com/jarslab/maven/babel/plugin/TranspilationInitializer.java

```diff
@@ -96,7 +96,12 @@ private Path determineTargetPath(final Path sourceFile)
     {
         final Path relativePath = getRelativePath(sourceFile);
         final String prefix = babelMojo.getPrefix() == null ? "" : babelMojo.getPrefix();
-        return babelMojo.getTargetDir().toPath().resolve(relativePath).resolve(prefix + sourceFile.getFileName());
+        
+        String targetFileName = prefix + sourceFile.getFileName();
+        if (babelMojo.getTargetFileExtension() != null) 
+        	targetFileName = targetFileName.replaceFirst("\\.[^./]+$", "."+babelMojo.getTargetFileExtension());
+        
+		return babelMojo.getTargetDir().toPath().resolve(relativePath).resolve(targetFileName);
```

Comment:
Minor thing, please use brackets for if statement.
Could you introduce some tests for this? Thanks!